src/gui/plugins/visualize_frustum/VisualizeFrustum.cc

```diff
@@ -0,0 +1,447 @@
+            if (!comp)
+            {
+              continue;
+            }
+            std::string childname = std::string(
+                            comp->Data());
+            if (nextstring.compare(childname) == 0)
+            {
+              parent = child;
+              foundChild = true;
+              if (i+1 == frustumURIVec.size()-1)
+              {
+                success = true;
+              }
+              break;
+            }
+          }
+          if (!foundChild)
+          {
+            gzerr << "The entity could not be found."
+                  << "Error displaying frustum visual" <<std::endl;
+            return;
+          }
+        }
+        if (success)
+        {
+          this->dataPtr->frustumEntity = parent;
+          this->dataPtr->frustumEntityDirty = false;
+        }
+      }
+    }
+  }
+
+  // Only update frustumPose if the frustumEntity exists and the frustum is
+  // initialized and the sensor message is yet to arrive.
+  //
+  // If we update the worldpose on the physics thread **after** the sensor
+  // data arrives, the visual is offset from the obstacle if the sensor is
+  // moving fast.
+  if (!this->dataPtr->frustumEntityDirty && this->dataPtr->initialized &&
+                                          !this->dataPtr->visualDirty)
+  {
+    this->dataPtr->frustumPose = worldPose(this->dataPtr->frustumEntity, _ecm);
+  }
+}
+
+//////////////////////////////////////////////////
+void VisualizeFrustum::OnTopic(const QString &_topicName)
+{
+  if (!this->dataPtr->topicName.empty() &&
+      !this->dataPtr->node.Unsubscribe(this->dataPtr->topicName))
+  {
+    gzerr << "Unable to unsubscribe from topic ["
+          << this->dataPtr->topicName <<"]" <<std::endl;
+  }
+
+  this->dataPtr->topicName = _topicName.toStdString();
+  std::lock_guard<std::mutex> lock(this->dataPtr->serviceMutex);
+
+  // Reset visualization
+  this->dataPtr->resetVisual = true;
+
+  // Create new subscription
+  if (!this->dataPtr->node.Subscribe(this->dataPtr->topicName,
+                            &VisualizeFrustum::OnScan, this))
+  {
+    gzerr << "Unable to subscribe to topic ["
+          << this->dataPtr->topicName << "]\n";
+    return;
+  }
+  this->dataPtr->visualDirty = false;
+}
+
+//////////////////////////////////////////////////
+void VisualizeFrustum::DisplayVisual(bool _value)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->serviceMutex);
+  this->dataPtr->frustum->SetVisible(_value);
+  gzerr << "Frustum Visual Display " << ((_value) ? "ON." : "OFF.")
+        << std::endl;
+}
+
+/////////////////////////////////////////////////
+void VisualizeFrustum::OnRefresh()
+{
+  // Clear
+  this->dataPtr->topicList.clear();
+
+  // Get updated list
+  std::vector<std::string> allTopics;
+  this->dataPtr->node.TopicList(allTopics);
+  for (auto topic : allTopics)
+  {
+    std::vector<transport::MessagePublisher> publishers;
+    this->dataPtr->node.TopicInfo(topic, publishers);
```

Comment:
One deprecation warning picked up by CI. This should fix it and make CI green:

```suggestion
    std::vector<transport::MessagePublisher> publishers;
    std::vector<transport::MessagePublisher> subscribers;
    this->dataPtr->node.TopicInfo(topic, publishers, subscribers);
```